openmdao/docs/openmdao_book/features/recording/case_recording_options.ipynb

```diff
@@ -73,6 +72,122 @@
     "|options_excludes         |      |  X   |      |       |"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "665ae681",
+   "metadata": {},
+   "source": [
+    "## Case Recorder Files\n",
+    "\n",
+    "When you instantiate a recorder, you must specify the filename to which the SQLite database of data is stored.\n",
+    "By default, this file will be stored under the output directory of the problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dd6f486e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import openmdao.api as om\n",
+    "\n",
+    "p = om.Problem(reports=None)\n",
+    "p.model.add_subsystem('exec', om.ExecComp('z = x + y'),\n",
+    "                       promotes_inputs=['x', 'y'], promotes_outputs=['z'])\n",
+    "\n",
+    "p.add_recorder(om.SqliteRecorder('prob_recorder.sql'))\n",
+    "\n",
+    "p.setup()\n",
+    "p.set_val('x', 3)\n",
+    "p.set_val('y', 5)\n",
+    "p.run_model()\n",
+    "p.record('first_case')\n",
+    "\n",
+    "p.set_val('x', 4)\n",
+    "p.set_val('y', 5)\n",
+    "p.run_model()\n",
+    "p.record('second_case')\n",
+    "\n",
+    "p.cleanup()\n",
+    "\n",
+    "# Now prob_recorder.sql should exist in the problem's outputs directory.\n",
+    "os.listdir(p.get_outputs_dir())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70d27e86",
+   "metadata": {
+    "tags": [
+     "remove-input",
+     "remove-output"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "assert((p.get_outputs_dir() / 'prob_recorder.sql').exists())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "489c88c2",
+   "metadata": {},
+   "source": [
+    "If the filename contains path separators (`/`), then OpenMDAO will assume that the recorder file path is not to be prepended with the problem's output directory.\n",
```

Comment:
I've struggled with this.

* `prob.get_outputs_dir()` isn't figured out until setup, the user applying this to a recorder before setup will encounter an exception. We could use some meta variable like `$OUTPUTS/myfile.sql` but that's asking the user to do something go get what should be, in my opinion, the default behavior of it going to the outputs directory.

* Specifying `foo/bar/baz.sql` would be erroneous because it's not just going to create those directories for you if they don't already exist.

Originally I thought it best to allow for paths so that the user could save the file wherever they wanted to. But I'm second guessing that.

1. If we let it be explicit, I worry we're going to get a lot of users who don't understand why their recorder file isn't going where they expect. I feel like that should be the default, but it should be overridable.  

2. Maybe we just force the user to specify a legal filename and insist that it goes in the outputs directory. This is probably the easiest from our side. No path separators would be allowed.

3. This all led me to the "if you give me a filename, it goes in outputs, otherwise we respect the path."